hypertrace-gradle-docker-java-application-plugin/src/main/java/org/hypertrace/gradle/docker/HypertraceDockerJavaApplicationPlugin.java

```diff
@@ -63,9 +63,13 @@ public void apply(Project target) {
   }
 
   private void addApplicationExtension(Project project) {
-    this.getHypertraceDockerExtension(project)
-        .getExtensions()
-        .create(EXTENSION_NAME, HypertraceDockerJavaApplication.class, project.getName());
+    HypertraceDockerJavaApplication extension =
+        this.getHypertraceDockerExtension(project)
+            .getExtensions()
+            .create(EXTENSION_NAME, HypertraceDockerJavaApplication.class, project.getName());
+
+    extension.javaVersion.convention(
+        getJavaExtension(project).getToolchain().getLanguageVersion().map(JavaVersion::toVersion));
```

Comment:
This is the only functional change - adjusting the convention (the default) to use the language version of the toolchain.